upkeep: update deps and deprecate java_version (#19)
buildbuddy-io/buildbuddy-toolchain: README.md

## Java support

Bazel provides support for Java toolchains out of the box.
You can enabled the Java toolchain with the following flags:

```

> --java_language_version=11

Reviewer: 11 is super old at this point, should we recommend 17 in the defaults?

Author: s/11/17/ sgtm. Applied

--tool_java_language_version=11
--java_runtime_version=remotejdk_11
--tool_java_runtime_version=remotejdk_11
```

> Available verions ar listed in [Bazel's User Manual](https://bazel.build/docs/user-manual#java-language-version)

Reviewer: ```suggestion
Available verions are listed in [Bazel's User Manual](https://bazel.build/docs/user-manual#java-language-version)
```

## Additional resources

- Check out our official documentation for [RBE Setup](https://www.buildbuddy.io/docs/rbe-setup)
- For more advanced use cases, check out Bazel's [bazel-toolchains repo](https://github.com/bazelbuild/bazel-toolchains) and the [docs on configuring C++ toolchains](https://docs.bazel.build/versions/master/tutorial/cc-toolchain-config.html).

> - Many thanks to the team at Grail who's [LLVM toolchain repo](https://github.com/bazel-contrib/toolchains_llvm) served as the basis for this repo.

Reviewer: It's no longer maintained by or associated with Grail, the original maintainer left them and handed this over to others (including me).

Author: Reworded to thanks the maintainers of the repo instead.

- Major props to the team at VSCO who's [toolchain repo](https://github.com/vsco/bazel-toolchains) paved the way for using LLVM as a Bazel toolchain.

## Other CC toolchains

> For advance users who want to write your own CC toolchain, here are some existing CC toolchains that you could use as reference:

Reviewer: ```suggestion
For advanced users who want to write their own CC toolchain, these existing CC toolchains that can serve as references:
```
